### PR TITLE
Add userspace config.h handling to build script

### PR DESCRIPTION
**Pull request by drashna**
The reason for this is that the user.h file doesn't get called in such a way that will modify QMK settings.  (ie, my settings were not being honored).    

Additionally, if I included the file in the config.h file, it would error out.
* with `#include "quantum.h"` would attempt to redefine everything in them, and error.
*without would not define `SAFE_RANGE` and stuff like "bool", erroring out. 

So this allows for you to use a config.h file in the userspace folder that is processed like the keyboard/keymap file.   Thus allowing for more consistent and global handling of settings.  

TL;DR: `<name>.h` is for user specific stuff, and `config.h` is for globally configuring keyboard settings

### CODE REVIEW

**Comment by jackhumbert** [comment]
Oops - conflicts! :)

**Comment by drashna** [comment]
Yup, because I had the code in my code PR, and not in the userspace config.h PR.

Fixed.

**Comment by jackhumbert** [comment]
Awesome :) thanks!
